homeassistant/components/tado/sensor.py

```diff
@@ -125,6 +125,13 @@ def unit_of_measurement(self):
         if self.zone_variable == "ac":
             return None
 
+    @property
+    def device_class(self):
+        """Return the device class."""
+        if self.zone_variable == "humidity":
+            return DEVICE_CLASS_HUMIDITY
+        return None
+
     @property
     def icon(self):
```

Comment:
Please don't set a custom icon when setting device class. Those entities already have specific icons in the frontend. Adjust the humidity case in the icon property.

Reply:
Thanks for pointing that out. I've removed it, and in the process did the same for temperature for consistency.